### PR TITLE
feat: aggregated balance UI

### PR DESCRIPTION
**Pull request by sahar-fehri**
## **Description**

Aggregated balance UI feature;

This PR fixed UI designs related to the aggregated balance feature;
It also fixes the popover behavior;
Users should be able to see the popover by default and should see it only once.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27161?quickstart=1)

## **Related issues**

Fixes:
Related:
https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4098-126757&node-type=instance&focus-id=4186-130770&m=dev
https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4496-20506&node-type=FRAME&m=dev

## **Manual testing steps**

1. After a new user is onboarded or upgraded, they should be able to see the aggregated balance popover.
2. Close the popover. Make sure you are not able to see the popover again even after turning on and off the new setting "show native token as main balance"

### **After**

https://github.com/user-attachments/assets/4c01541e-0d77-46c3-8a21-98dc3114e11d

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: ui/components/ui/dropdown/dropdown.scss
Comment (sahar-fehri) [inline ui/components/ui/dropdown/dropdown.scss:6]: We made this change on component level
cc @amandaye0h

File: ui/components/ui/icon-button/icon-button.scss
Comment (sahar-fehri) [inline ui/components/ui/icon-button/icon-button.scss:27]: We made this change on component level
cc @amandaye0h

File: ui/components/ui/text-field/text-field.component.js
Comment (sahar-fehri) [inline ui/components/ui/text-field/text-field.component.js:209]: This component is deprecated, i created a [ticket](https://consensyssoftware.atlassian.net/browse/MMASSETS-393) to refactor the search bar in setting to use the `TextFieldSearch` component instead;
With @amandaye0h  we thought it makes sense to make these changes globally to all components

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [inline ui/components/app/wallet-overview/coin-overview.tsx:260]: Hey @georgewrmarshall 👋 
We have these [designs](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4098-126757&node-type=instance&focus-id=4186-130770&m=dev) show the arrow of popover at the very start of the component, even though i am using `PopoverPosition.BottomStart` , it still does not quite match the designs;
Did i maybe misused the component?
cc @amandaye0h 
Screenshot 2024-09-16 at 12 31 17

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [inline ui/components/app/wallet-overview/coin-overview.tsx:263]: Hello again @georgewrmarshall 👋 
We also have this color in [figma](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4098-126757&node-type=instance&focus-id=4186-130770&m=dev) that looks a bit darker than the 
`backgroundColor={BackgroundColor.overlayAlternative} `

Sorry, I wasn't quite sure about the solution you suggested 🙏

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/coin-overview.tsx:280]: Then we can remove these individual color delclarations
```suggestion
```

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/coin-overview.tsx:299]:
```suggestion
```

**Review by georgewrmarshall — COMMENTED** [review]
Hey @sahar-fehri, this is looking great! Left some suggestions to achieve the reverse theme for the `Popover` component

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx:263]: Thank you @georgewrmarshall!
That was the trick 🙏 
Looks good now

cc @amandaye0h 

Screenshot 2024-09-16 at 23 59 47

Screenshot 2024-09-16 at 23 58 57

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx:214]: Yes here is the updated component after using buttonLink!

cc @amandaye0h 

https://github.com/user-attachments/assets/dbe771ef-b14d-43c9-a6b8-7b55f70af404

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: fixed here [05e181b](https://github.com/MetaMask/metamask-extension/pull/27161/commits/05e181ba23ac86dfe206593d64ae6bee843bf87d)

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: fixed here [05e181b](https://github.com/MetaMask/metamask-extension/pull/27161/commits/05e181ba23ac86dfe206593d64ae6bee843bf87d)

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: fixed [05e181b](https://github.com/MetaMask/metamask-extension/pull/27161/commits/05e181ba23ac86dfe206593d64ae6bee843bf87d)

File: ui/components/component-library/button-icon/button-icon.tsx
Comment (sahar-fehri) [inline ui/components/component-library/button-icon/button-icon.tsx:27]: Adding support for Xs size for `ButtonIconSize`

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [inline ui/components/app/wallet-overview/coin-overview.tsx:214]: `ButtonLink` component was not applying variant: `TextVariant.bodyMdMedium` i sent in textProps;
Im passing className instead to overrida text css changes;

File: ui/pages/home/index.scss
Comment (sahar-fehri) [reply ui/pages/home/index.scss]: Padding removed

File: ui/components/component-library/button-icon/button-icon.tsx
Comment (garrettbear) [reply ui/components/component-library/button-icon/button-icon.tsx:27]: Might get some pushback on adding Xs size.

Will need to ask design first @Akatori-Design @justincheungcon

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (garrettbear) [inline ui/components/app/wallet-overview/coin-overview.tsx:213]: The goal should be to use design system components out of the box without making modifications. If the designs aren't align, then maybe we should take a look and discuss with designer on why they chose to customize further. 

As for the `TextVariant`, I was able to get it working without the need of the className. 

```suggestion
            textProps={{ variant: TextVariant.bodyMdMedium }}
```

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (garrettbear) [inline ui/components/app/wallet-overview/coin-overview.tsx:207]: Can remove this styling as well since the designs should be using design system ButtonLink component and not needing the additional spacing

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (garrettbear) [reply ui/components/app/wallet-overview/coin-overview.tsx:262]: The challenge here is that the Popover component has a lot of responsive behavior vs static design, but you could try `position=PopoverPosition.BottomStart` to see if that works.

Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx:262]: Hi @garrettbear ! Thnx for checking this out 🙏  The current code (which matches the screenshots) is already using `position={PopoverPosition.BottomStart} `
I was only able to modify the arrow position by doing what you see in video ☝️

File: ui/components/app/wallet-overview/index.scss
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/index.scss:46]: Using static font styles I would consider a code smell we should endeavour to be using the Text component and props. All component library components have prop objects that allow access to the children components so you should be able to achieve this withough using these styles

```suggestion
```

File: ui/components/app/wallet-overview/coin-buttons.tsx
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/coin-buttons.tsx:428]: non-blocking: Would be great to get a storybook file for this component so we can view and work on it in isolation

File: ui/components/component-library/button-icon/button-icon.types.ts
Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/button-icon.types.ts:9]: In the design it looks like `Sm` is being used. If it's ok I'm going to block this PR until these changes have been reverted 

Screenshot 2024-09-17 at 1 44 57 PM

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (georgewrmarshall) [reply ui/components/app/wallet-overview/coin-overview.tsx:214]: Why not this?
```
<ButtonLink
endIconName={IconName.Export}
onClick={handlePortfolioOnClick}
as="a"
data-testid="portfolio-link"
className="wallet-overview__portfolio_text"
>
{t('portfolio')}
</ButtonLink>
```


https://github.com/user-attachments/assets/2e5390ed-2131-4da4-a226-0aaad4c6c6d7

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey Sahar, this is looking great, I'm just blocking this because I think there is an unnecessary update to a design system component that we may not want to include until we get confirmation. Also there are some other comments that should help maintain consistency and clean code. Happy to pair also!

File: ui/components/component-library/button-icon/button-icon.types.ts
Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/button-icon.types.ts:9]: ### After
Using `ButtonIconSize.Sm` seems to align with the design size 

Screenshot 2024-09-17 at 1 58 54 PM
Screenshot 2024-09-17 at 1 58 58 PM

Comment (sahar-fehri) [reply ui/components/component-library/button-icon/button-icon.types.ts:9]: Hi @georgewrmarshall ! 
Thanks for the feedback 🙏  im using this in the popover [here](https://github.com/MetaMask/metamask-extension/pull/27161/files#diff-7c9c4f1322c8b8a3ca7308d1bc14aaece2feab78b6c280bda58e1de7e1143e62R286)
This is how it looks with the current code (Xs) VS how it looks with Sm;

Thought that even though in Figma it says "Sm"; with the code "Xs" looks better;
cc @amandaye0h 

Screenshot 2024-09-17 at 22 57 06
Screenshot 2024-09-17 at 22 56 30

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: removed wallet-overview__portfolio_button and used `margin: 2` in `endIconProps` instead

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx:214]: This works too! 🙏  just adding marginLeft to keep '8px' between portfolio text and portfolio export icon
```
          <ButtonLink
            endIconName={IconName.Export}
            endIconProps={{
              marginLeft: 2,
            }}
            onClick={handlePortfolioOnClick}
            as="a"
            data-testid="portfolio-link"
            className="wallet-overview__portfolio_text"
          >
            {t('portfolio')}
          </ButtonLink>
```
but i see in the suggestion you are using `className="wallet-overview__portfolio_text"`,
Initially i wanted to use textProps to pass my props; but they were not applied
 textProps={{ variant: TextVariant.bodyMdMedium }}

image


One other solution would be to not use ButtonLink component and do something that looks like this with few tweaks
```
      <Box
            onClick={handlePortfolioOnClick}
            data-testid="portfolio-link"
          >
            <Text variant={TextVariant.bodyMd} color={TextColor.primaryDefault}>
              {t('portfolio')}
            </Text>

            <Icon
              size={IconSize.Sm}
              name={IconName.Export}
              color={IconColor.primaryDefault}
            />
          </Box>
```

File: ui/components/app/wallet-overview/index.scss
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/index.scss]: Removed here [52a8669](https://github.com/MetaMask/metamask-extension/pull/27161/commits/52a8669072e05aef09347df60e9ae1459dd543d3)

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: Thanks @garrettbear ! Fixed here [52a8669](https://github.com/MetaMask/metamask-extension/pull/27161/commits/52a8669072e05aef09347df60e9ae1459dd543d3)

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx:201]: Resolving this; this is fixed here [52a8669](https://github.com/MetaMask/metamask-extension/pull/27161/commits/52a8669072e05aef09347df60e9ae1459dd543d3) 🙏

File: ui/components/app/wallet-overview/coin-buttons.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-buttons.tsx:428]: Makes sense, added here [fa835b1](https://github.com/MetaMask/metamask-extension/pull/27161/commits/fa835b12d05ca8fe3994d5316913d9ed4e7b2bdc)

File: ui/components/component-library/button-icon/button-icon.types.ts
Comment (sahar-fehri) [reply ui/components/component-library/button-icon/button-icon.types.ts]: I have eventually reverted this change here [9fe85dd](https://github.com/MetaMask/metamask-extension/pull/27161/commits/9fe85ddcd7b5a0e029eba222c1990023e74a5123)
Now using : `size={ButtonIconSize.Sm}`

**Review by amandaye0h — APPROVED** [review]

File: ui/components/app/wallet-overview/coin-buttons.stories.js
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/coin-buttons.stories.js:37]: Excellent! 🔥 

Screenshot 2024-09-18 at 9 23 47 AM

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (georgewrmarshall) [reply ui/components/app/wallet-overview/coin-overview.tsx:201]: It seems like there's a bug or it's not very intuitive. When `ButtonLink` uses `size="inherit"` or `size="auto"` (e.g. `<ButtonLink size={ButtonLinkSize.Inherit}>`, `<ButtonLink size={ButtonLinkSize.Auto}>`), the CSS applies `inherit` to the font styles, which prevents the style utility props from working. cc @garrettbear

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (georgewrmarshall) [inline ui/components/app/wallet-overview/coin-overview.tsx:268]: Can we follow responsive typography here and avoid locking the font-size to 12px? It should scale up on larger screens but stay at 12px on smaller screens.

https://github.com/user-attachments/assets/df120ff5-06a5-430b-ba8e-29eecee8eab4

**Review by georgewrmarshall — COMMENTED** [review]
Hey Sahar, feel free to merge to unblock, but just a heads up—using static CSS for `font-size` and `font-weight` is a code smell for the design system. We shouldn’t need to use those at all. We might even want to create a stylelint rule to prevent their usage. It's more likely the design isn't using the correct Figma library for Typography. If you remove all static declarations for `font-size` and `font-weight`, it will be design system approved! cc @amandaye0h

File: ui/components/app/wallet-overview/coin-overview.tsx
Comment (sahar-fehri) [reply ui/components/app/wallet-overview/coin-overview.tsx]: Thanks @georgewrmarshall ! fixed here [5e720c6](https://github.com/MetaMask/metamask-extension/pull/27161/commits/5e720c6ca6d693fd7006e113861c9c3e0a054ade)
I removed ` className="balance-popover__text"`

File: ui/components/ui/dropdown/dropdown.scss
Comment (georgewrmarshall) [reply ui/components/ui/dropdown/dropdown.scss:6]: It may be worth doing a check of all the instances that this might effect to ensure there are no visual regressions

File: ui/components/component-library/button-icon/button-icon.types.ts
Comment (georgewrmarshall) [reply ui/components/component-library/button-icon/button-icon.types.ts]: Resolved

**Review by georgewrmarshall — APPROVED** [review]
This is looking great! Thank you for your diligence and openness to design system suggestions. I noticed a few small consistency updates were included, such as the settings headings. Breaking these changes into a separate PR would help with the review process, allow design system engineers to focus on details, and reduce the size of this PR. Just a suggestion for the future!

File: ui/selectors/selectors.js
Comment (gambinish) [inline ui/selectors/selectors.js:1562]: Can we remove this log?

Comment (gambinish) [reply ui/selectors/selectors.js:1562]: Also fine with you handling this on your feature branch so we don't block here.

File: app/scripts/controllers/app-state.js
Comment (gambinish) [inline app/scripts/controllers/app-state.js:77]: Would be cool one day if we could scope all of these "onboarding" related state variables within a single object, like `onboardingState: {}` to make it easier to understand which vars are related to onboarding vs what's related to the root level metamask state (unrelated to onboarding)

**Review by gambinish — APPROVED** [review]
Gonna approve this as is. As we discussed, there's an issue where the popover doesn't show on Testnet (works as expected on Mainnet). May want to verify behavior with Product here.

Feel free to address this, along with my other `console.log` nit on your feature branch. If you end up making changes here just ping me on slack so I can re-approve.

**Comment by bergeron** [comment]
I think fullscreen looks a little worse than before, being left aligned with all that space.  That said, I don't have a perfect suggestion for what it should do instead.

Screenshot 2024-09-19 at 3 27 34 PM

**Comment by gambinish** [comment]
> I think fullscreen looks a little worse than before, being left aligned with all that space. That said, I don't have a perfect suggestion for what it should do instead.
> 
> Screenshot 2024-09-19 at 3 27 34 PM

I also noticed this. This is in the [designs](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4098-126568&node-type=instance&focus-id=4186-130770&m=dev) though. I agree though

**Comment by sahar-fehri** [comment]
> > I think fullscreen looks a little worse than before, being left aligned with all that space. That said, I don't have a perfect suggestion for what it should do instead.
> > Screenshot 2024-09-19 at 3 27 34 PM
> 
> I also noticed this. This is per the [designs](https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=4098-126568&node-type=instance&focus-id=4186-130770&m=dev) spec though. I also felt like it was a bit jarring (maybe I'll get used to it)
> 
> Maybe the monetization buttons can get flexed to the right of the window, rather than be vertical aligned...

I think Amanda and Hester are aligned on this; this is not the final state but its an okay state for couple of weeks

**Pull request merged**
